packages/svelte-parse/src/main.ts

```diff
@@ -0,0 +1,1219 @@
+					//@ts-ignore
+					current_node.position = { start: place(), end: {} };
+				continue;
+			}
+
+			// "/" or  ">" (for void tags) put us in a terminal state
+			if (
+				char === SLASH ||
+				(char === CLOSE_ANGLE_BRACKET &&
+					is_void_element((current_node as SvelteElement).tagName))
+			) {
+				set_state(State.IN_CLOSING_SLASH, true);
+				(current_node as BaseSvelteTag).selfClosing = true;
+				if (char === SLASH) chomp();
+				continue;
+			}
+
+			if (char === CLOSE_ANGLE_BRACKET) {
+				set_state(State.PARSE_CHILDREN, true);
+				chomp();
+				//@ts-ignore
+				if (generatePositions) current_node.position.end = place();
+				continue;
+			}
+
+			if (char === SPACE || char === TAB || char === LINEFEED) {
+				chomp();
+				continue;
+			}
+		}
+
+		if (current_state === State.IN_SHORTHAND_ATTR) {
+			if (char === CLOSE_BRACE) {
+				(current_node as Property).value[0].value = (current_node as Property).name;
+				pop_state();
+				pop_node();
+				chomp();
+				continue;
+			}
+
+			(current_node as Property).name += value[index];
+			chomp();
+			continue;
+		}
+
+		// we are expecting the tag to close completely here
+		if (current_state === State.IN_CLOSING_SLASH) {
+			// ignore ws
+			if (char === SPACE || char === TAB || char === LINEFEED) {
+				chomp();
+				continue;
+			}
+			// we closed successfully, end the parse
+			if (char === CLOSE_ANGLE_BRACKET) {
+				chomp();
+				// @ts-ignore
+				if (generatePositions) current_node.position.end = place();
+				break;
+			}
+
+			// DANGER ZONE - something went wrong
+		}
+
+		// we are parsing a property name
+		if (current_state === State.IN_ATTR_NAME) {
+			let s;
+			// " ", "\n", "/" or ">" => shorthand boolean attr
+
+			if (
+				(s = char) === SPACE ||
+				s === TAB ||
+				s === LINEFEED ||
+				s === SLASH ||
+				s === CLOSE_ANGLE_BRACKET
+			) {
+				(current_node as Property).shorthand = 'boolean';
+				pop_state();
+				pop_node();
+				continue;
+			}
+
+			// ":" => directive
+			if (char === COLON) {
+				//@ts-ignore
+				(current_node as Directive).type = 'svelteDirective';
+				(current_node as Directive).specifier = '';
+				set_state(State.IN_DIRECTIVE_SPECIFIER, true);
+				chomp();
+				continue;
+			}
+
+			if (char === PIPE) {
+				chomp();
+				const _n = { value: '', type: 'modifier' };
+				if (generatePositions)
+					//@ts-ignore
+					_n.position = { start: place(), end: [] };
+				(current_node as Directive).modifiers.push(_n as Literal);
+				push_node(_n);
+				set_state(State.IN_ATTR_MODIFIER, true);
+				continue;
+			}
+
+			if (char === EQUALS) {
+				set_state(State.IN_ATTR_VALUE, true);
+				chomp();
+				continue;
+			}
+
+			// process the token and chomp, everything is good
+			(current_node as Property).name += value[index];
+			chomp();
+			continue;
+		}
+
+		// att values can be quoted or unquoted
+		if (current_state === State.IN_ATTR_VALUE) {
+			// ignore whitespace it is valid after `=`
+			let s;
```

Comment:
same here (and in a few more places below. Requesting new variable space for read-only copy